pkg/controller/core.oam.dev/v1alpha2/application/apply.go

```diff
@@ -172,17 +182,24 @@ func (h *appHandler) apply(ctx context.Context, appRev *v1beta1.ApplicationRevis
 	ac.SetOwnerReferences(owners)
 	h.FinalizeAppRevision(appRev, ac, comps)
 
+	if h.autodetect {
+		// TODO(yangsoon) autodetect is temporarily not implemented
+		return fmt.Errorf("helm mode component doesn't specify workload")
```

Comment:
So the helm based component still can't work in this PR?

Reply:
The helm-based component still works. This error will be recorded in application CR, to tell the reason why the traits attached to the helm mode component were failed to work.